src/btree/bt_cursor.c

```diff
@@ -512,12 +512,36 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
     return (0);
 }
 
+/*
+ * __wt_btcur_evict_reposition --
+ *     Try to evict the page and reposition the cursor on the saved key.
+ */
+int
+__wt_btcur_evict_reposition(WT_CURSOR_BTREE *cbt)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    cursor = &cbt->iface;
+    session = CUR2S(cbt);
+
+    if (session->txn->isolation == WT_ISO_SNAPSHOT && !WT_SESSION_IS_CHECKPOINT(session) &&
+      F_ISSET_ATOMIC_16(cbt->ref->page, WT_PAGE_FORCE_EVICTION)) {
```

Comment:
We can check the number of hazard point here and if it is larger than 1, we should fall not try to evict.

Reply:
I don't think there is currently a function that does what you're suggesting. The closest function is probably `__wt_hazard_check`. 
Do you think it will be worth it to create another function that checks for all sessions in a connection if they point to the same page?

Reply:
@agorrod @Siddhartha8899 I think we should create that function to check that. Releasing the cursor, failing to do the eviction, and then replacing the cursor is expensive.

Reply:
The new function would be similar to __wt_hazard_check but only fails the check if the hazard point is from another session.